node/test/go/aws-lambda-sdk/integration.test.js

```diff
@@ -0,0 +1,225 @@
+'use strict';
+
+const { expect } = require('chai');
+
+const path = require('path');
+const log = require('log').get('test');
+const { TracePayload } = require('@serverless/sdk-schema/dist/trace');
+const cleanup = require('./lib/cleanup');
+const createCoreResources = require('./lib/create-core-resources');
+const basename = require('./lib/basename');
+const getProcessFunction = require('../../lib/get-process-function');
+const resolveOutcomeEnumValue = require('../../utils/resolve-outcome-enum-value');
+const resolveNanosecondsTimestamp = require('../../utils/resolve-nanoseconds-timestamp');
+const normalizeEvents = require('../../utils/normalize-events');
+const spawn = require('child-process-ext/spawn');
+const resolveTestVariantsConfig = require('../../lib/resolve-test-variants-config');
+const resolveDirZipBuffer = require('../../utils/resolve-dir-zip-buffer');
+
+const fixturesDirname = path.resolve(__dirname, '../../../../go/packages/slslambda/fixtures');
+
+for (const name of ['TEST_EXTERNAL_LAYER_FILENAME']) {
+  // In tests, current working directory is mocked,
+  // so if relative path is provided in env var it won't be resolved properly
+  // with this patch we resolve it before cwd mocking
+  if (process.env[name]) process.env[name] = path.resolve(process.env[name]);
+}
+
+describe('Go: integration', function () {
+  this.timeout(120000);
+  const coreConfig = {};
+
+  function buildZipFile(functionName, architecture) {
+    return async () => {
+      const script = path.resolve(__dirname, 'build-go-binary.sh');
+      const moduleDir = path.resolve(fixturesDirname, functionName);
+      await spawn(script, [moduleDir, architecture]);
+      const outputDir = path.resolve(moduleDir, 'build', architecture);
+      const zipFile = await resolveDirZipBuffer(outputDir);
+      const isArm64 = architecture === 'arm64';
+      return {
+        Runtime: isArm64 ? 'provided.al2' : 'go1.x',
+        Architectures: isArm64 ? ['arm64'] : ['x86_64'],
+        Code: { ZipFile: zipFile },
+      };
+    };
+  }
```

Comment:
Ideally if zip file generation is done via some script in context of Go package. So we can build the layer manually, being in context of the Go package (and don't have to run Node.js integration test for that)

Reply:
Go SDK won't have a layer. This part is just preparing the Lambda deployment package (building the code and zipping the binary). I'll move the shell script to `go` directory but there will be no need to run this script in another context than the integration test.

Reply:
Ok, so this script bundles fixtures (as located in go package) with the SDK (?) Then it's indeed specific to testing, but as we keep fixtures in Go package, I believe it's best if it's put somewhere aside them (?)

Reply:
Moved to `go/packages/slslambda/scripts/build-go-binary.sh`